File: project_euler/Problem062/Problem062.py
import math

class Problem062:
    def solution(self) -> None:
        target, cubes, index, pivot = 5, {}, 5, 0
        while True:
            value = index ** 3
            digits, count = self.__get_digits(value), int(math.log10(value))
            if pivot != count:
                cubes.clear()
                pivot = count
            else:
                if digits in cubes and len(cubes[digits]) == target - 1:
                    print(cubes[digits][0])
                    return
            if digits not in cubes:
                cubes[digits] = []
            cubes[digits].append(value)
            index += 1

    def __get_digits(self, value: int) -> int:
        result = 0
        while value:
            result += int(math.pow(10, value % 10))
            value //= 10
        return result

File: project_euler/Problem062/Problem062.java
package project_euler;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Problem062 {
    public void solution() {
        int target = 5;
        Map<Long, List<Long>> cubes = new HashMap<>();
        int index = 5;
        int pivot = 0;
        while (true) {
            long value = (long) Math.pow(index, 3);
            long digits = getDigits(value);
            int count = (int) Math.log10(value);
            if (pivot != count) {
                cubes.clear();
                pivot = count;
            } else {
                if (cubes.containsKey(digits) && cubes.get(digits).size() == target - 1) {
                    System.out.println(cubes.get(digits).get(0));
                    return;
                }
            }
            if (!cubes.containsKey(digits)) {
                cubes.put(digits, new ArrayList<>());
            }
            cubes.get(digits).add(value);
            ++index;
        }
    }

    private long getDigits(long value) {
        long result = 0L;
        while (0L != value) {
            result += (long) Math.pow(10, value % 10);
            value /= 10;
        }
        return result;
    }
}
